lib/luanetfilter.c

```diff
@@ -43,6 +73,67 @@ static void luanetfilter_release(void *private)
 	lunatik_putobject(luaxtable->runtime);
 }
 
+/* Register the target with the netfilter module. The caller must push the table on the stack.
+ * Pops the table from the stack both on success and failure.
+ * Pushes the error string on the stack on failure and returns -1.
+ * */
+static int handle_xtable_reg(lua_State *L, luanetfilter_xtable_t *luaxtable, LUANETFILTER_MODE mode)
+{
+	int base = lua_gettop(L);
+	int type;
+
+	const char *name = NULL;
+	u_int8_t revision = 0;
+	unsigned short family = 0;
+	unsigned short proto = 0;
+
+	type = lua_type(L, base);
+	if (type == LUA_TNIL){
+		pr_err("[luanetfilter] no ops found, assuming nil\n");
+		return 0;
+	} else if (type != LUA_TTABLE){
+		lua_pushstring(L, "invalid ops type, expected table or nil");
+		goto err;
+	}
+
+	luanetfilter_xtable_op(L, base, "name", lua_isstring);
+	name = lua_tostring(L, -1);
```

Comment:
you can't persist strings in this way.. take a look at the manual..

Reply:
I had this just for debugging, the storage to registry will come after this.